packages/table/src/HeaderRow/HeaderRow.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { PropsWithChildren, ReactElement, ReactNode } from 'react';
+
+import { HeaderCell } from '../Cell';
+
+import { HeaderRowProps } from './HeaderRow.types';
+
+const HeaderRow = ({
+  children,
+  ...rest
+}: PropsWithChildren<HeaderRowProps>) => {
+  return (
+    <tr {...rest}>
+      {React.Children.map(children, (child: ReactNode, index: number) => {
```

Comment:
in the storybook it looks like the consumer is responsible for rendering HeaderCell components inside of HeaderRow components is this not the case?

Reply:
No because HeaderCell needs to know the rendered order of the cell to determine the padding. If we removed this, we would need the user to pass in the index themselves but I figured we'd prioritize DX here. 

One other approach we could take is to use CSS `:first-of-type`, but it would also have to know if the cell is expandable, or is a checkbox cell, etc. That would take a bit more changes and I think that might be a nice small future improvement?

Reply:
@TheSonOfThomp Good call! Adding that.